packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -24,33 +24,87 @@ export const displayTxResult = (
   if (typeof displayContent === "bigint") {
     try {
       const asNumber = Number(displayContent);
-      if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
-        return asNumber;
+      if (asText) {
+        if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
+          return asNumber;
+        } else {
+          return "Ξ" + formatEther(displayContent);
+        }
       } else {
-        return "Ξ" + formatEther(displayContent);
+        return <NumberDisplay value={displayContent} />;
       }
     } catch (e) {
       return "Ξ" + formatEther(displayContent);
     }
   }
 
-  if (typeof displayContent === "string" && isAddress(displayContent)) {
-    return asText ? displayContent : <Address address={displayContent} />;
+  if (typeof displayContent === "string") {
+    if (isAddress(displayContent)) {
+      return asText ? displayContent : <Address address={displayContent} />;
+    }
+
+    if (isHex(displayContent)) {
+      return displayContent; // don't add quotes
+    }
   }
 
   if (Array.isArray(displayContent)) {
-    const mostReadable = (v: DisplayContent) =>
-      ["number", "boolean"].includes(typeof v) ? v : displayTxResultAsText(v);
-    const displayable = JSON.stringify(displayContent.map(mostReadable), replacer);
-
-    return asText ? (
-      displayable
-    ) : (
-      <span style={{ overflowWrap: "break-word", width: "100%" }}>{displayable.replaceAll(",", ",\n")}</span>
-    );
+    return asText ? JSON.stringify(displayContent, replacer) : <TupleDisplay value={displayContent} />;
+  }
+
+  if (typeof displayContent === "object") {
```

Comment:
Notice the two structs : 
![Screenshot 2024-05-29 at 2 37 14 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/095369ab-58ec-4e93-92b6-9bab86238211)


While doing a write transaction we get a receipt, which is also an object more precisely nested object :(

<details>

<summary>
Example : 
</summary>

![Screenshot 2024-05-29 at 2 53 08 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/4a534a26-d6fc-4f1d-a502-29220bdba4aa)


</details>

Maybe here : 

https://github.com/scaffold-eth/scaffold-eth-2/blob/c670c1f0d841b4a4e7ddf74d67e62c154d099d81/packages/nextjs/app/debug/_components/contract/TxReceipt.tsx#L43

We check if txResult?.txResult exist then we pass it ?